### PR TITLE
feat(storage): filtering

### PR DESCRIPTION
**Pull request by jurajpiar**
This PR will allow the UI to filter storage offers by the average price of their billing plans.
To achieve this it calculates the mean price when in the `BillingPlanSet` event handler and stores it in a field `averagePrice`.

This PR also adds changes to the storage hooks that allow consumer to filter the storage records.

Resolves cache requirements for [RMKT-247](https://rsklabs.atlassian.net/jira/software/projects/RMKT/boards/60?assignee=5e84f3015b08780c1c4127b0&selectedIssue=RMKT-247)

Blocking [RMKT-29](https://rsklabs.atlassian.net/browse/RMKT-29)

### CODE REVIEW

**Review by AuHau — APPROVED** [review]
LGTM!

**Review by nduchak — APPROVED** [review]
🥇

**Comment by AuHau** [comment]
@jurajpiar since you are still adding new commits I have marked the PR as draft. Once you are finished please mark it ready for review.

Also please keep the PRs to the point and try to focus only on one thing if possible and create different PRs for different changes.

File: src/services/storage/hooks/offers.hooks.ts [hunk 1]
```diff
@@ -22,7 +23,53 @@ export default {
...
+
+            if (averagePrice) {
+              context.params.sequelize.where.averagePrice = {
+                [Op.between]: [averagePrice.min, averagePrice.max]
+              }
+            }
+
+            if (totalCapacity) {
+              context.params.sequelize.where.totalCapacity = {
+                [Op.between]: [totalCapacity.min, totalCapacity.max]
```
Comment (nduchak) [inline src/services/storage/hooks/offers.hooks.ts:62]: Is this filter works???
`totalCapacity` is gype of `string`

Comment (jurajpiar) [reply src/services/storage/hooks/offers.hooks.ts:62]: no no of course not, as as you pointed out the totalCapacity is a type string.

Comment (jurajpiar) [reply src/services/storage/hooks/offers.hooks.ts]: Now it should work fine :) Thanks for finding the casting process @nduchak.

Comment (nduchak) [reply src/services/storage/hooks/offers.hooks.ts]: Looks cool. Thank you 👍

Comment (nduchak) [reply src/services/storage/hooks/offers.hooks.ts]: Hmmm.. 
But this `totalCapacity.min` and `totalCapacity.max` coming from request? It's a possible SQL injection
You probably should sanitize or add some validation that these fields are actually a number

**Pull request merged**